libraries/redcore/api/hal/helper.php

```diff
@@ -958,19 +959,86 @@ public static function getPrimaryKeysFromFields($xmlElement)
 			{
 				$fieldAttributes = self::getXMLElementAttributes($field);
 
-				if (self::isAttributeTrue($field, 'isPrimaryField'))
+				if (($primaryKeys && self::isAttributeTrue($field, 'isPrimaryField'))
+					|| !$primaryKeys)
 				{
-					$primaryKeys[$fieldAttributes['name']] = $fieldAttributes;
+					$fields[$fieldAttributes['name']] = $fieldAttributes;
 				}
 			}
 		}
 
 		// If there are no primary keys defined we will use id field as default
-		if (empty($primaryKeys))
+		if (empty($fields))
```

Comment:
I would make another check here to see if we are looking for primaries only... because if we are not then we do not have to have any field